### PR TITLE
ENH: Support for WRF >= 3.7

### PR DESCRIPTION
**Pull request by andreas-h**
WRF changed its `configure` script in version 3.7.  This PR checks for WRF version and handles both cases (version <= 3.6 and version >= 3.7)

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyblocks/w/wrf.py [hunk 1]
```diff
@@ -116,10 +118,12 @@ def configure_step(self):
         build_type_option = None
         self.comp_fam = self.toolchain.comp_family()
         if self.comp_fam == toolchain.INTELCOMP:  #@UndefinedVariable
-            build_type_option = "Linux x86_64 i486 i586 i686, ifort compiler with icc"
+            build_type_option = ("INTEL\ \(ifort\/icc\)" if LooseVersion(self.version) >= LooseVersion('3.7')
+                                 else "Linux x86_64 i486 i586 i686, ifort compiler with icc")
```
Comment (boegel) [inline easybuild/easyblocks/w/wrf.py]: @andreas-h I would reformat this to this, for readability:

```python
if LooseVersion(self.version) >= LooseVersion('3.7'):
    build_type_option = "INTEL\ \(ifort\/icc\)"
else:
    build_type_option = "Linux x86_64 i486 i586 i686, ifort compiler with icc"
```

same below

File: easybuild/easyblocks/w/wrf.py [hunk 2]
```diff
@@ -133,7 +137,10 @@ def configure_step(self):
             raise EasyBuildError("Unknown build type: '%s'. Supported build types: %s", bt, known_build_types)
 
         # fetch option number based on build type option and selected build type
-        build_type_question = "\s*(?P<nr>[0-9]+).\s*%s\s*\(%s\)" % (build_type_option, bt)
+        if LooseVersion(self.version) >= LooseVersion('3.7'):
+            build_type_question = "\s*(?P<nr>[0-9]+)\.\ \(%s\).*%s" % (bt, build_type_option)
```
Comment (boegel) [inline easybuild/easyblocks/w/wrf.py]: would it make sense to include an example as a comment above? same below

**Comment by boegel** [comment]
@andreas-h this looks very good, thanks a lot for looking into this!

Do you have an updated easyconfig file as well, that can be used as a testcase for this change?

**Comment by andreas-h** [comment]
@boegel I have an easyconfig file which I suppose is correct, but it currently does not build successfully on my system.  But I guess this isn't easybuild related, as I also have trouble building directly (i.e., without EasyBuild).

I uploaded the easyconfig file I'm using here: https://gist.github.com/andreas-h/de2af52d1348a00b308997717986655d but not all dependencies are in the easybuild-easyconfigs repo yet, I guess.

I uploaded the build output showing my problems here: https://gist.github.com/andreas-h/c9ec2df5dea7a0ac18bed2e3643ad2e7 but I guess that's is off-topic for this issue ;)

**Comment by boegel** [comment]
@andreas-h that log doesn't correspond to the `foss` easyconfig you showed?

The problem in the log is that `gcc` is being combined with these `$CFLAGS`, which are for the Intel compilers:

```
== 2016-04-26 17:42:10,258 environment.py:97 INFO Environment variable CFLAGS set to -O2 -xHost -ftz -fp-speculation=safe -fp-model source
```

**Comment by boegel** [comment]
@andreas-h ah, sorry, you have both easyconfigs in your gist... are you hitting similar problems when using `foss`?

**Comment by boegel** [comment]
@andreas-h one extra change is required for recent versions of WRF, cfr. https://github.com/andreas-h/easybuild-easyblocks/pull/1

**Comment by boegel** [comment]
@andreas-h I opened a PR with an easyconfig for WRF 3.8, see https://github.com/hpcugent/easybuild-easyconfigs/issues/2963

Works for me using your updated easyblock, thanks!

**Comment by boegel** [comment]
tested with both existing WRF easyconfigs, and the one in https://github.com/hpcugent/easybuild-easyconfigs/pull/2963, so good to go

Thanks @andreas-h!

**Pull request merged**